### DISCUSSION TITLE
Loading Fonts through javascript

### DISCUSSION THREAD

**Discussion by KokoDoko** [excaliburjs/Excalibur#2606] [Ideas]
I have found that if I load fonts using javascript, I don't have to use polling or eventlisteners to be sure that fonts from the CSS file have loaded. It makes loading fonts a lot easier. I add this code to `resources.js`, and I remove the font loading code from `index.html` and `style.css`. The actual font file has to be in the project.

```javascript
import fontFile from "../css/PressStart2P-Regular.ttf"

const font = new FontFace("coolFont", `url(${fontFile})`)
document.fonts.add(font)
font.load()
```
It seems that the listener is not needed. It would be cool if this code could be added to the official `Loader` class as a `Resource`?
```javascript
font.load().then(()=>{console.log("the font is loaded"})
```
You can use it like this
```javascript
this.scoreText = new Text({
            text: 'Hello Henk',
            font: new Font({
                family: 'coolFont',
                size: 20,
            }),
 })
```

**Comment by eonarheim** [comment]
@KokoDoko I think this is a great idea! I'd love if this was added as a Resource! I would support that change!

**Comment by KokoDoko** [comment]
To reply to my own post, I found this resource that also makes font loading easier: https://fontfaceobserver.com
